### PR TITLE
Prune deps, comment deny.toml

### PR DESCRIPTION
**Pull request by pchickey**
Follow up on #2519 

1. Run `cargo update` to get the latest deps.
1. Eliminate a dep on an old `wasmparser` by turning off cargo features we weren't using.
1. Comment on all skipped duplicate deps on why they are an issue.
1. Upgrade to wasmparser 0.71 in tree. Leave some new simd instructions as `wasm_unsupported!` in cranelift-wasm.

The duplicate deps fall into two categories:
1. transitive deps pull in different versions. Some stuff, like `cfg-if` and `env_logger`, are so popular that it would be very hard to motivate the whole rust ecosystem to use the very latest.
1. frequently-updated crates we maintain use, like `wasmparser` and `wast`, are used directly and also transitively via dependencies which live in other repos. It would create toil to keep these consistent all the time. We can put occasional effort into keeping these updated, but it isn't worth being perfect.

### CODE REVIEW

File: deny.toml [hunk 1]
```diff
@@ -37,12 +37,11 @@ deny = []
...
-    { name = "humantime" },
-    { name = "memoffset" },
-    { name = "wasmparser" },
-    { name = "wast" },
+    { name = "ansi_term" }, # transitive dependencies only
+    { name = "cfg-if" }, # transitive dependencies use 0.1
+    { name = "env_logger" }, # pretty_env_logger and file-per-thread-logger depend on  0.7
+    { name = "humantime" }, # caused by env_logger
+    { name = "getrandom" }, # rand not updates to 0.2 yet
+    { name = "wasmparser" }, # rapid development, some 0.70 and 0.71
```
Comment (alexcrichton) [inline deny.toml:45]: With the dependency through `object` we should control all the dependencies on `wasmparser`, so I think this should be fixable by updating everything in-tree to 0.71?

File: deny.toml [hunk 2]
```diff
@@ -37,12 +37,11 @@ deny = []
...
-    { name = "cfg-if" },
-    { name = "env_logger" },
-    { name = "getrandom" },
-    { name = "humantime" },
-    { name = "memoffset" },
-    { name = "wasmparser" },
-    { name = "wast" },
+    { name = "ansi_term" }, # transitive dependencies only
+    { name = "cfg-if" }, # transitive dependencies use 0.1
+    { name = "env_logger" }, # pretty_env_logger and file-per-thread-logger depend on  0.7
```
Comment (bjorn3) [inline deny.toml:42]:
```suggestion
    { name = "env_logger" }, # pretty_env_logger and file-per-thread-logger depend on 0.7
```

File: deny.toml (hunk 1 above)
Comment (pchickey) [reply deny.toml:45]: Fixed fb78025

Comment (pchickey) [reply deny.toml:45]: Note: had to leave some new simd instructions as wasm_unsupported! in cranelift-wasm.

**Review by alexcrichton — APPROVED** [review]
Looks great!

**Pull request merged**